src/analysis.zig

```diff
@@ -25,15 +25,18 @@ resolved_callsites: std.AutoHashMapUnmanaged(Declaration.Param, ?TypeWithHandle)
 resolved_nodes: std.HashMapUnmanaged(NodeWithUri, ?TypeWithHandle, NodeWithUri.Context, std.hash_map.default_max_load_percentage) = .{},
 /// used to detect recursion
 use_trail: NodeSet = .{},
+/// handle of the doc where the request originated
+root_handle: *DocumentStore.Handle = undefined,
```

Comment:
Could you make this an optional pointer. The analysis backend should not need to know about what a LSP request is.

Reply:
Maybe I've worded it poorly but it's simply the handle of the first node any lookup happens on, ie given a `some_import.dependency_var.other` it's the handle `some_import` node belongs to, so this can work:
`some_import` belongs to a handle (A)  with a build file, and
`dependency_var` belongs to a handle (B)  with it's own build file, yet it contains an `@import` which references a module of handle A.

Given that zig allows it, the real proper solution would be to keep a chain of handles when doing lookup and walk it up in such cases.

Anyhow, you have the big picture of the analyzer, so please make any changes you deem necessary.

Reply:
I do understand the reasoning behind the root handle.
I just don't think that initializing an `Analyser` should force specifying a root handle because there may not be a root handle in certain situations. One example would be when handling a `workspace/symbol` request where no root handle exists.

Reply:
Got it ! .. and as always, open to suggestions when it comes to naming variables :face_exhaling: